### DISCUSSION TITLE
Connecting to in memory sqlite database

### DISCUSSION THREAD

**Discussion by fearcon-eap** [sfu-db/connector-x#171] [Q&A]
Hey,

is there a way to connect to an "in-memory" sqlite database?

```
sql_url = 'sqlite://:memory:'
df = cx.read_sql(sql_url,QUERY)
```
Something in the likes of:

`source = sqlite3.connect(':memory:')`

**Comment by wangxiaoying** [comment]
Hi @fearcon-eap , we only support connect to sqlite on disk files for now. Will take a look at this feature. Thanks for the suggestion!

**Comment by wangxiaoying** [comment]
Hi @fearcon-eap , sorry for the delayed reply. I checked [sqlite in-memory database](https://www.sqlite.org/inmemorydb.html) and it seems like "in-memory" database's life time is the same with the connection:

> The database ceases to exist as soon as the database connection is closed. Every :memory: database is distinct from every other. So, opening two database connections each with the filename ":memory:" will create two independent in-memory databases.

However, in connectorx we establish our own connection and do not hold it after return the data. We also only support select query instead of execution (e.g. create table, insert values). So it seems like there is no use case for us supporting sqlite's "in-memory" database.

Please feel free to let us know if you have other thoughts or suggestions on this feature!
